content/source/docs/glossary.html.md

```diff
@@ -0,0 +1,940 @@
+
+- [Wikipedia: JSON](https://en.wikipedia.org/wiki/Json)
+- [Terraform docs: JSON Configuration Syntax](/docs/configuration/syntax-json.html)
+
+## Locking
+
+[locking]: glossary.html#locking
+[lock]: glossary.html#locking
+
+-> Terraform Enterprise
+
+The ability to prevent new [runs][] from starting in a given [workspace][]. Workspaces are automatically locked while a run is in progress, and can also be manually locked.
+
+Some other Terraform [backends][] can also lock state during runs.
+
+## Log
+
+[log]: glossary.html#log
+[logs]: glossary.html#log
+
+The text-based output of actions taken within a [run][]. For example, the output of running `terraform plan`.
+
+## Module
+
+[module]: glossary.html#module
+[modules]: glossary.html#module
+
+Also "Terraform module".
+
+A self-contained collection of Terraform [configurations][] that manages a collection of related infrastructure resources.
+
+Other Terraform configurations can _call_ a module, which tells Terraform to manage any resources described by that module.
+
+Modules define [input variables][] (which the calling module can set values for) and [output values][] (which the calling module can reference in [expressions][]).
+
+- [Terraform docs: Modules](/docs/modules/index.html)
+
+## OAuth
+
+[oauth]: glossary.html#oauth
+
+An open standard for token-based authorization between applications on the internet.
+
+Terraform Enterprise uses OAuth to connect your organization to your [VCS provider][]. Generally takes an `id` and `secret` from your VCS provider to give access to TFE and allow it to download configuration from the provider.
+
+- [TFE docs: VCS Integration](/docs/enterprise/vcs/index.html)
+
+## OAuth Client
+
+[oauth client]: glossary.html#oauth-client
+[oauth clients]: glossary.html#oauth-client
+
+-> Terraform Enterprise
+
+The set of configuration that a TFE organization needs in order to connect to a specific [VCS provider][].
+
+An OAuth client needs an [OAuth token][] in order to actually access data belonging to a user or organization in that VCS provider. The client can be created with an existing token for that VCS provider (when created via TFE's API), or it can be created with the details TFE needs in order to request a token. Requesting a token requires a user to click through and approve access with their VCS provider account.
+
+- [TFE docs: VCS Integration](/docs/enterprise/vcs/index.html)
+- [TFE API docs: OAuth Clients](/docs/enterprise/api/oauth-clients.html)
+
+## OAuth Token
+
+[oauth token]: glossary.html#oauth-token
+[oauth tokens]: glossary.html#oauth-token
+
+-> Terraform Enterprise
+
+The access token that a TFE organization needs in order to access data belonging to a user or organization in a specific [VCS provider][]. The token is granted by the VCS provider, and allows access with whatever permissions that VCS provider assigns to the token.
```

Comment:
Ah, good call. I can't remember that, but that's not to say you didn't say it to me. 😅

I'm gonna do this one directly in the file, since it spans a few paragraphs.